src/core/filesystem.cc

```diff
@@ -1269,7 +1319,19 @@ Status
 DestroyFileFolder(const std::string& path)
 {
   FileSystem* fs;
-  RETURN_IF_ERROR(GetFileSystem(path, &fs));
+  // If path represents local temporary file then must be S3
+  if (path.rfind("/tmp/file", 0) == 0) {
+#ifdef TRTIS_ENABLE_S3
+    Aws::SDKOptions options;
+    Aws::InitAPI(options);
+    static S3FileSystem s3_fs(options);
+    fs = &s3_fs;
```

Comment:
In fact we actually don't care about the file system type, we will delete the folder if and only if the path is to our designated tmp path. We just need to check if the path is the designated path (we know it must be in local file system), then use the `LocalFileSystem::DestroyFileFolder()` to do the right thing.

Reply:
You just need to move what is in the `S3FileSystem::DestroyFileFolder()` over to `LocalFileSystem`. Currently it is a little bit strange because "S3FileSystem" is deleting local files...

Reply:
Fixed